### ISSUE TITLE
Heisenbug: Meteor logs out randomly, no error messages

### ISSUE THREAD

**Issue by pcuci** [meteor-useraccounts/core#557]
This is useless complaining as I can't replicate the bug. Maybe some of you have had a similar issue and can guess how to dissect the problem...

I get logged out for no apparent reason. I log-in again, and it now works. I sign out, then sign-in again, and for a few moments I'm logged-in, then it kicks me out. `Meteor.user()` shows `null` - but, but I just logged in!

Any way to turn on AccountsTemplates logging on or some other mechanism to figure out the root issue here?

**Comment by rteslaru** [comment]
I'm just seeing the exact same behavior. It only happens if I logout and then log back in immediately (as in 1-2 seconds). If I wait, say, 10+ seconds between logins, it doesn't happen.

**Comment by splendido** [comment]
Does this happen also in case you log-out and then log-in from the browser's console?

...could this be linked to the new [DDPRateLimiter](https://docs.meteor.com/#/full/ddpratelimiter)?

**Comment by rteslaru** [comment]
Interesting, I didn't know about it. It sounds like it might very well be linked to that. I see this when logging in/out quickly, for various testing purposes.

I'll be able to check in a few hours and will post back.

Will also test with `Accounts.removeDefaultRateLimit()` to see if that fixes it.

**Comment by splendido** [comment]
Yeah, I had to call `Accounts.removeDefaultRateLimit()` for the `splendido:accounts-meld` tests to work again ;-)  See [these lines](https://github.com/splendido/meteor-accounts-meld/blob/master/tests/accounts-meld_tests.js#L16-L17)

**Comment by splendido** [comment]
@pcuci could it be the same for you?

**Comment by rteslaru** [comment]
FWIW, I can't reproduce the bug anymore. Heisenbug indeed. Will keep an eye out for it.

**Comment by jbrozena22** [comment]
+1, we've had this issue occur on two separate meteor projects. Can't reproduce either.

**Comment by rteslaru** [comment]
Started again. `Accounts.removeDefaultRateLimit()` does not seem to help.

**Comment by brianmcdonough** [comment]
This bug was observable for one user and could be reproduced over and over for that user, but without error or any useful information in the logs. 

In my setup, (hosted on Modulus, db on Compose) after another user logs in, the bug resolves itself and the first user can log in and stay logged in. 

I hope this information can help point to the source, but seems to be the kind of bug that wants to stay hidden.

**Comment by nicholaai** [comment]
+1 Experiencing this same problem. Just like @rteslaru "It only happens if I logout and then log back in immediately (as in 1-2 seconds). If I wait, say, 10+ seconds between logins, it doesn't happen."

**Comment by laran** [comment]
Calling Accounts.removeDefaultRateLimit did nothing to help. I'm running integration tests around my login code which run multiple tests with various username/password combinations. I need to run them with a setTimeout of 2000 ms at the end to prevent the timeout.

I saw Accounts.removeDefaultRateLimit mentioned here:
I called http://experimentsinmeteor.com/meteor-testing-with-velocity-and-jasmine-part-2/

... and tried it.

I've tried calling it at startup in a Meteor.ifServer block. I've tried defining a server method that calls it and then calling the server method before my test suite. I also tried calling the server method from the client before every test. None of these things worked. Every one hit the rate limiter. Using setTimeout is the only way I've found to get around the rate limiter.

Either I'm doing something dumb (99% likely), or it doesn't seem like the disable is working properly. Would love to see some sample code for how to call it properly.

**Comment by eliasthompson** [comment]
+1

**Comment by jacobjuul** [comment]
+1 having this error also. It seems to be fine with facebook logins, but not email logins.

**Comment by HammadSiddiqui** [comment]
+1

**Comment by robinBanquo** [comment]
+1

**Comment by eybarta** [comment]
has anyone found a solution to this problem? 
I'm running Meteor 1.6 and randomly many active users are getting kicked out of the system.. seeing no errors in the galaxy logs..

**Comment by inkiltie** [comment]
Problem still there - Meteor 1.6.0.1

**Comment by claytonmarinho** [comment]
+1

**Comment by pagesrichie** [comment]
Having the same error..

**Comment by evolross** [comment]
I just had a user get this tonight during a live stream of using our app. We're on Meteor 1.9.3. And I've seen this for what seems like years. Even in development. The app will just randomly log out sometimes, with plenty of activity. It's VERY intermittent.

**Comment by evolross** [comment]
So this was probably my misunderstanding Accounts.config's `loginExpirationInDays`. Apparently it logs the user out no matter what even if they have activity. It only _resets_ if the users logs out and back in. I had this value set to `2`. So users get a nice log out after two days. I guess this is meant for much longer durations as a "well don't let them be logged in forever" kind of safety mechanism.

I'm pretty sure this is the cause of the issue for me and maybe others here. [Meteor doesn't handle the inactivity](https://github.com/meteor/meteor-feature-requests/issues/119#issuecomment-364748281) for you.
